clients/src/main/java/org/apache/kafka/clients/consumer/internals/PrototypeAsyncConsumer.java

```diff
@@ -546,6 +619,22 @@ public ConsumerRecords<K, V> poll(long timeout) {
         throw new KafkaException("method not implemented");
     }
 
+    private void maybeWakeup() {
+        if (this.closed)
+            throw new IllegalStateException("This consumer has already been closed.");
+
+        log.debug("Raising WakeupException in response to user wakeup");
```

Comment:
Not sure this is super useful. Maybe we could add the wakeup exception so that at least we get the stack trace?

Reply:
This is actually in the original wakeup call see: (I forgot why I left it out of the getAndSet block, but I'll move it into the block.
```
        if (!wakeupDisabled.get() && wakeup.get()) {
            log.debug("Raising WakeupException in response to user wakeup");
            wakeup.set(false);
            throw new WakeupException();
        }
```

Reply:
Ok. Not a big deal either way.